aspnetcore/security/enforcing-ssl.md

```diff
@@ -10,45 +10,71 @@ ms.technology: aspnet
 >
->* Not listen on HTTP.
->* Close the connection with status code 400 (Bad Request) and not serve the request.
+> * Not listen on HTTP.
+> * Close the connection with status code 400 (Bad Request) and not serve the request.
 
 <a name="require"></a>
 ## Require HTTPS
 
 ::: moniker range=">= aspnetcore-2.1"
-We recommend all ASP.NET Core web apps call `UseHttpsRedirection` to redirect all HTTP requests to HTTPS. If `UseHsts` is called in the app, it must be called before `UseHttpsRedirection`.
+
+We recommend all ASP.NET Core web apps call HTTPS Redirection Middleware ([UseHttpsRedirection](/dotnet/api/microsoft.aspnetcore.builder.httpspolicybuilderextensions.usehttpsredirection)) to redirect all HTTP requests to HTTPS.
 
 The following code calls `UseHttpsRedirection` in the `Startup` class:
 
-[!code-csharp[sample](enforcing-ssl/sample/Startup.cs?name=snippet1&highlight=13)]
+[!code-csharp[](enforcing-ssl/sample/Startup.cs?name=snippet1&highlight=13)]
 
+The following code calls [AddHttpsRedirection](/dotnet/api/microsoft.aspnetcore.builder.httpsredirectionservicesextensions.addhttpsredirection) to configure middleware options:
 
-The following code:
+[!code-csharp[](enforcing-ssl/sample/Startup.cs?name=snippet2&highlight=14-99)]
 
-[!code-csharp[sample](enforcing-ssl/sample/Startup.cs?name=snippet2&highlight=14-99)]
+The preceding highlighted code:
 
-* Sets `RedirectStatusCode`.
+* Sets [HttpsRedirectionOptions.RedirectStatusCode](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.redirectstatuscode).
 * Sets the HTTPS port to 5001.
 
-::: moniker-end
+The following mechanisms set the port automatically:
+
+* The middleware can discover the ports via [IServerAddressesFeature](/dotnet/api/microsoft.aspnetcore.hosting.server.features.iserveraddressesfeature) when the following conditions apply:
+  - Kestrel or HTTP.sys is used directly with HTTPS endpoints (also applies to running the app with Visual Studio Code's debugger).
+  - Only **one HTTPS port** is used by the app.
+* Visual Studio is used:
+  - IIS Express has HTTPS enabled.
+  - *launchSettings.json* sets the `sslPort` for IIS Express.
+
+> [!NOTE]
+> When an app is run behind a reverse proxy (for example, IIS, IIS Express), `IServerAddressesFeature` isn't available. The port must be manually configured. When the port isn't set, requests aren't redirected.
+
+The port can be configured by setting the:
+
+* `ASPNETCORE_HTTPS_PORT` environment variable.
+* URL with the `ASPNETCORE_URLS` environment variable.
+* `http_port` host configuration key (for example, via *hostsettings.json* or a command line argument).
+* [HttpsRedirectionOptions.HttpsPort](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.httpsport). See the preceding example that shows how to set the port to 5001.
 
+If no port is set:
+
+* Requests aren't redirected.
+* The middleware logs a warning.
+* The middleware is disabled.
```

Comment:
"Requests aren't redirected." and "The middleware is disabled." are redundant. The first one's better.